examples/easycat/easycat_example.cc

```diff
@@ -17,14 +19,51 @@ using namespace kickcat;
 
 int main(int argc, char* argv[])
 {
-    if (argc != 2)
+    if (argc != 3 and argc != 2)
     {
-        printf("usage: ./test NIC\n");
+        printf("usage redundancy mode : ./test NIC_nominal NIC_redundancy\n");
+        printf("usage no redundancy mode : ./test NIC_nominal\n");
         return 1;
     }
 
-    auto socket = std::make_shared<Socket>();
-    Bus bus(socket);
+
+    auto reportRedundancy = []()
+    {
+        printf("Redundancy has been activated due to loss of a cable \n");
+    };
+
+    auto socketNominal = std::make_shared<Socket>();
+
+    std::shared_ptr<AbstractSocket> socketRedundancy;
```

Comment:
Parse all command line arguments in one section and then use the result instead of mixing parsing and bus creation.